caput/time.py

```diff
@@ -863,16 +1081,71 @@ def ephemeris(self):
     def reload(self):
         """Reload the Skyfield data regardless of the `expire` setting."""
 
-        exp_val = self.load.expire
-        self.load.expire = True
-
-        try:
-            self.timescale
-            self.ephemeris
-        finally:
-            self.load.expire = exp_val
+        self.load.download("finals2000A.all")
+        self.load.download(self._ephemeris_name)
 
 
 # Set up a module local Skyfield wrapper for time conversion functions in this
 # module to use.
 skyfield_wrapper = SkyfieldWrapper()
+
+
+def _solve_all(f, x0, x1, dx, skip_increasing=False, skip_decreasing=False, **kwargs):
+    """Find all roots of function f, within the interval [x0, x1].
+
+    To find all the roots we need an estimate of the spacing of the roots. This is
+    specified by the parameter `dx`. If this is too large, roots may be missed.
+
+    Parameters
+    ----------
+    f : callable
+        A numpy vectorized function which returns a single value f(x).
+    x0, x1 : float
+        The start and end of the interval to search.
+    dx : float
+        An interval known to be less than the spacing between the closest roots.
+    skip_increasing, skip_decreasing : bool, optional
+        Skip roots where the gradient is positive (or negative).
+    **kwargs
+        Passed to `scipy.optimize.brentq`. `xtol`, `rtol` and `maxiter` are probably
+        the most useful.
+
+    Returns
+    -------
+    roots : np.ndarray[np.float64]
+        The values of the roots found.
+    increasing : np.ndarray[bool]
+        If the gradient at the root is positive.
+    """
+    # Form a grid of points to find intervals to search over
+    x_init = np.linspace(x0, x1, int(np.ceil((x1 - x0) / dx)), endpoint=True)
+    f_init = f(x_init)
+
+    roots = []
+    increasing = []
+
+    # Search through intervals
+    for xa, xb, fa, fb in zip(x_init[:-1], x_init[1:], f_init[:-1], f_init[1:]):
+
+        # Entries are the same sign, so there is no solution in between.
+        # NOTE: we need to deal with the case where one edge might be an exact root,
+        # hence the strictly greater than 0.0
```

Comment:
Does `brentq` accept one edge being the root? Or did you mean to deal with this case sepparately and forgot?

Reply:
It does indeed. You only need to take care of it when you're trying to detect regions which contain a solution.